python/astro_metadata_translator/tests.py

```diff
@@ -149,8 +150,26 @@ def assertObservationInfoFromYaml(self, file, dir=None, check_wcs=True,  # noqa:
             inconsistent.
         """
         header = read_test_file(file, dir=dir)
-        self.assertObservationInfo(header, filename=file, check_wcs=check_wcs,
-                                   wcs_params=wcs_params, **kwargs)
+
+        try:
+            self.assertObservationInfo(header, filename=file, check_wcs=check_wcs,
+                                       wcs_params=wcs_params, **kwargs)
+        except AssertionError as e:
+            raise AssertionError("ObservationInfo derived from an dict-like "
+                                 "type is inconsistent.") from e
+
+        astropy_header = Header()
+        for key, val in header.items():
+            values = val if isinstance(val, list) else [val]
+            for v in values:
+                astropy_header[key] = v
```

Comment:
There reason I wanted to use `astropy_header.append((key, v))` was that it would result in multiple card images if there were multiple list values. As written currently I think it just replaces each one with the caveat that I assume COMMENT and HISTORY are intercepted and turned into appends.

Reply:
It does only work for COMMENTS which is incidentally the only field I made a quick test with. 

```
a = {"SIMPLE": True, "NAXIS": 1, "FOO": [2, 3], "COMMENT": ["A ", "list"]}
header = Header()
for key, val in a.items():
    values = val if isinstance(val, list) else [val]
    for v in values:
        header[key] = v

# >>> header
# SIMPLE  =                    T                                                  
# NAXIS   =                    1                                                  
# FOO     =                    3                                                                                                 
# COMMENT A                                                                       
# COMMENT list           
```

Reply:
What I'm trying to say is that `append()` would work everywhere.

Reply:
So:
```python
from astropy.io.fits import Header
a = {"SIMPLE": True, "NAXIS": 1, "FOO": [2, 3], "COMMENT": ["A ", "list"]}
header = Header()
for key, val in a.items():
    values = val if isinstance(val, list) else [val]
    for v in values:
        header.append((key, v))

print(repr(header))
```
gives me:
```
SIMPLE  =                    T                                                  
NAXIS   =                    1                                                  
FOO     =                    2                                                  
FOO     =                    3                                                  
COMMENT A                                                                       
COMMENT list      
```
which I think is exactly what we want.

Reply:
Yes, exactly, it was more of a "I wasn't thorough" comment. Thanks!